src/routing/approved.rs

```diff
@@ -160,6 +160,24 @@ impl Approved {
             .clone())
     }
 
+    /// Returns the latest known public key of the section with `prefix`.
+    pub fn section_key(&self, prefix: &Prefix) -> Option<&bls::PublicKey> {
+        if prefix == self.section.prefix() || prefix.is_extension_of(self.section.prefix()) {
+            Some(self.section.chain().last_key())
+        } else {
+            self.network.key_by_prefix(prefix).or_else(|| {
+                if self.is_elder() {
+                    // We are elder - the first key is the genesis key
+                    Some(self.section.chain().first_key())
```

Comment:
what if we dont know that prefix section?
maybe better remain as return with None , instead of replace with the genesis key, in case the prefix is not our current prefix ?

Reply:
Because this doesn't return the current key of that section, but the last one we know. If we don't know the section at all then the last known key is the genesis key.

Reply:
bit confused here.
the function name is `section_key(prefix)` , indicating the key is for the section of prefix.
meanwhile here we use `our section's first_key(gensis)` to return

Reply:
It could have been called `last_known_section_key` or something like that. Decided to keep the name short as there are no other key-retrieval functions this could be confused with. Plus the doc comment should explain it.

> meanwhile here we use our section's first_key(gensis) to return

Thing is that "our section's first key" is also every other section's first key.